### PR TITLE
PhpdocAlignFixer - fix for whitespace in type

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4879

Pinging @fabpot, @Seldaek, @sstok, @GrahamCampbell, @keradus (as authors) and @enumag (as issue reporter) for review.

### CODE REVIEW

File: src/Fixer/Phpdoc/PhpdocAlignFixer.php [hunk 1]
```diff
@@ -90,19 +90,21 @@ public function configure(array $configuration = null)
         $tagsWithoutNameToAlign = array_diff($this->configuration['tags'], $tagsWithNameToAlign, $tagsWithMethodSignatureToAlign);
         $types = [];
 
+        $type = '\S(?:[^<\s]|<[^>]*>)*';
```
Comment (julienfalque) [inline src/Fixer/Phpdoc/PhpdocAlignFixer.php:93]: Maybe better using `PhpCsFixer\DocBlock\Annotation::REGEX_TYPES` somehow?

Comment (kubawerlos) [reply src/Fixer/Phpdoc/PhpdocAlignFixer.php:93]: It gives me:
```
Invalid PCRE pattern "/^(?P<indent>(?: {2}|\t)*) \* @(?:(?P<tag>param)\s+(?P<hint>(?:
    # <simple> is any non-array, non-generic, non-alternated type, eg `int` or `\Foo`
    # <array> is array of <simple>, eg `int[]` or `\Foo[]`
    # <generic> is generic collection type, like `array<string, int>`, `Collection<Item>` and more complex like `Collection<int, \null|SubCollection<string>>`
    # <type> is <simple>, <array> or <generic> type, like `int`, `bool[]` or `Collection<ItemKey, ItemVal>`
    # <types> is one or more types alternated via `|`, like `int|bool[]|Collection<ItemKey, ItemVal>`
    (?<types>
        (?<type>
            (?<array>
                (?&simple)(\[\])*
            )
            |
            (?<simple>
                [@$?]?[\\\w]+
            )
            |
            (?<generic>
                (?&simple)
                <
                    (?:(?&types),\s*)?(?:(?&types)|(?&generic))
                >
            )
        )
        (?:
            \|
            (?:(?&simple)|(?&array)|(?&generic))
        )*
    )
    )?)\s+(?P<var>(?:&|\.{3})?\$\S+)|(?P<tag2>return|throws|type|var)\s+(?P<hint2>(?:
    # <simple> is any non-array, non-generic, non-alternated type, eg `int` or `\Foo`
    # <array> is array of <simple>, eg `int[]` or `\Foo[]`
    # <generic> is generic collection type, like `array<string, int>`, `Collection<Item>` and more complex like `Collection<int, \null|SubCollection<string>>`
    # <type> is <simple>, <array> or <generic> type, like `int`, `bool[]` or `Collection<ItemKey, ItemVal>`
    # <types> is one or more types alternated via `|`, like `int|bool[]|Collection<ItemKey, ItemVal>`
    (?<types>
        (?<type>
            (?<array>
                (?&simple)(\[\])*
            )
            |
            (?<simple>
                [@$?]?[\\\w]+
            )
            |
            (?<generic>
                (?&simple)
                <
                    (?:(?&types),\s*)?(?:(?&types)|(?&generic))
                >
            )
        )
        (?:
            \|
            (?:(?&simple)|(?&array)|(?&generic))
        )*
    )
    )?))(?:\s+(?P<desc>\V*))\s*$/ux": (code: 1) Compilation failed: quantifier does not follow a repeatable item at offset 18 (version: 10.34 2019-11-21)
```

**Comment by sstok** [comment]
Note. I helped with a really old version of this fixer (before tokens were used) and have almost no knowledge the current version 👍

**Comment by enumag** [comment]
What's the current status of this?

File: tests/Fixer/Phpdoc/PhpdocAlignFixerTest.php [hunk 2]
```diff
@@ -1272,4 +1272,19 @@ public function provideInvalidPhpdocCases()
             ],
         ];
     }
+
+    public function testTypesContainingWhitespace()
```
Comment (GrahamCampbell) [inline tests/Fixer/Phpdoc/PhpdocAlignFixerTest.php:1276]: Could you add another test for something like `callable(string): bool`?

Comment (GrahamCampbell) [reply tests/Fixer/Phpdoc/PhpdocAlignFixerTest.php:1276]: Example from the wild: https://github.com/php-http/client-common/blob/master/src/Plugin.php#L27.

Comment (enumag) [reply tests/Fixer/Phpdoc/PhpdocAlignFixerTest.php:1276]: ping @kubawerlos

Comment (kubawerlos) [reply tests/Fixer/Phpdoc/PhpdocAlignFixerTest.php:1276]: @enumag the support for callables is here: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4914 - after this is merged I will continue with this PR.

**Review by GrahamCampbell — APPROVED** [review]

**Comment by keradus** [comment]
Thank you @kubawerlos.

**Pull request merged**